### ISSUE TITLE
included examples

### ISSUE THREAD

**Issue by toastxc** [linebender/xilem#174]
I tried out both examples (hello and taffy) on Windows 10 and they looked, interesting.

I think it would make more sense to create examples that are a bit better looking, if the project is at the stage where this is possible

**Comment by lazulkia** [comment]
Same expectation.
Many great projects have a wealth of examples, and some interesting examples may also spark new improvements. Sometimes the examples are more of a learning experience than the documentation.

List the examples I expect:
1, Show GIF images
2, Create a virtual keyboard (showing almost all keyboard-mouse interactions)
3, A menu window. Has all the characteristics of a menu, but it is a separate window. Similar:
![20240408085704](https://github.com/linebender/xilem/assets/158443534/ebd3fff9-2cc0-45da-96bc-e68dabba55cb)

4, Display a Markdown file

**Comment by toastxc** [comment]
That'd be good
also, if WASM is possible with this library, a website example is a good idea.

**Comment by DJMcNab** [comment]
Thank you for your interest! This is definitely on the agenda, however, we're still working on lower-level parts of the stack, to enable this.
As discussed in [This Month in Xilem, March 2024](https://linebender.org/blog/tmix-03/), an objective for the next month is to get Xilem into a state where it can be demonstrated at RustNL.

Unfortunately, work of this kind on the Xilem side is largely blocked on the implementation of [rfcs#1](https://github.com/linebender/rfcs/blob/main/rfcs/0001-masonry-backend.md), which I believe is making good progress.

Those suggested examples are interesting, although I would like to ask for some clarification:
1) Is there any particular reason the image has to be in GIF format? I'm not sure that is a very strong example - I suspect SVG might be more suitable, as that makes better use of our vector graphics foundations?
2) I'm not sure what you're suggesting. Do you mean just a text field?
3) We are definitely aiming to do menus properly, but unfortunately this is made more difficult than it should be by winit.
4) Displaying markdown is definitely a useful property to have, but doing an actual integration properly would be quite challenging - this is due to markdown also permitting HTML tags, which means you need to implement large parts of a browser to make it work

On the comment of WASM:
This is quite a deep subject, but for the moment I would suggest using `xilem_web` instead of `xilem` for WASM. This is because it is implausible and/or impossible to integrate the Xilem widget layer into the web in a manner which is sympathetic to the needs of our users (e.g. in terms of accessibility).
I personally hope we can create a unified API on top of both, but that is very much future work. Our focus in the near term is on the non-web cases.

**Comment by DJMcNab** [comment]
This issue isn't really actionable as-is. We definitely have the intention to make better examples.
